ios/Errors.swift

```diff
@@ -1,55 +1,667 @@
+                code = rn
+            } else {
+                code = RNErrorCode.UNEXPECTED_SDK_ERROR.rawValue
+            }
+            nativeErrorCode = String(nsError.code)
+        } else {
+            let nonStripeErrorMapping = mapNonStripeError(nsError: nsError)
+            code = nonStripeErrorMapping.code
+            nativeErrorCode = nonStripeErrorMapping.nativeErrorCode
+        }
+        
+        return ErrorInformation(
+            isStripeError: isStripeError,
+            code: code,
+            nativeErrorCode: nativeErrorCode,
+            errorName: isStripeError ? ErrorConstants.stripeErrorName : ErrorConstants.nonStripeErrorName
+        )
+    }
+    
+    /// Maps non-Stripe errors to the unified error structure.
+    ///
+    /// Handles errors that don't originate from the Stripe Terminal SDK by creating a generic
+    /// error code and native error identifier.
+    ///
+    /// - Parameter nsError: The non-Stripe NSError to map
+    /// - Returns: A tuple containing the React Native error code and formatted native error code
+    private class func mapNonStripeError(nsError: NSError) -> (code: String, nativeErrorCode: String) {
+        return (
+            code: RNErrorCode.UNEXPECTED_SDK_ERROR.rawValue,
+            nativeErrorCode: "\(nsError.domain):\(nsError.code)"
+        )
+    }
+    
+    // MARK: - Top-level Field Extraction
+    
+    /// Extracts API error information from NSError and adds it to the result dictionary.
+    ///
+    /// Maps iOS NSError data to the unified apiError structure, including decline code extraction
+    /// via KVC for Confirm*Error types.
+    ///
+    /// - Parameters:
+    ///   - nsError: The NSError containing API error information
+    ///   - result: The result dictionary to populate (modified in place)
+    private class func extractApiErrorToTopLevel(from nsError: NSError, to result: inout [String: Any]) {
+        let className = String(describing: type(of: nsError))
+        var apiError: [String: Any] = [:]
+        
+        // Extract declineCode from Confirm*Error via KVC
+        if className.contains("ConfirmPaymentIntentError") || 
+           className.contains("ConfirmSetupIntentError") ||
+           className.contains("SCPConfirmPaymentIntentError") ||
+           className.contains("SCPConfirmSetupIntentError") {
+            
+            if let declineCode = nsError.value(forKey: "declineCode") as? String {
+                apiError[ErrorConstants.apiErrorDeclineCodeKey] = declineCode
+            }
+        }
+        
+        if let failureReason = nsError.userInfo[ErrorConstants.scpStripeAPIFailureReason] as? String {
+            apiError[ErrorConstants.apiErrorMessageKey] = failureReason
+        } else {
+            // Use localized description as fallback message
+            apiError[ErrorConstants.apiErrorMessageKey] = nsError.localizedDescription
+        }
+
+        apiError[ErrorConstants.apiErrorCodeKey] = String(nsError.code)
```

Comment:
I don't think this is correct. The nsError.code will always be SCPErrorStripeAPIError for API errors.

https://stripe.dev/stripe-terminal-ios/docs/Enums/SCPError.html#/c:@E@SCPError@SCPErrorStripeAPIError

this should be the value from SCPErrorKeyStripeAPIErrorCode if I'm following correctly.